source/material_model/visco_plastic.cc

```diff
@@ -925,9 +922,12 @@ namespace aspect
           dislocation_creep.initialize_simulator (this->get_simulator());
           dislocation_creep.parse_parameters(prm, std::make_shared<std::vector<unsigned int>>(n_phase_transitions_for_each_composition));
 
-          // FK viscosity parameters
-          frank_kamenetskii_rheology.initialize_simulator (this->get_simulator());
-          frank_kamenetskii_rheology.parse_parameters(prm);
+          // Frank Kamenetskii viscosity parameters
+          if (viscous_flow_law == frank_kamenetskii)
+            {
+              frank_kamenetskii_rheology.initialize_simulator (this->get_simulator());
+              frank_kamenetskii_rheology.parse_parameters(prm);
```

Comment:
The changes above this block are obviously correct, but I don't understand this. It should always be ok to call `initialize_simulator()` and `parse_parameters()` assuming that `frank_kamenetskii` is a valid object. Why do you need to guard these calls?

Reply:
Or are you saying "we only need this object if the user selected FK rheology, and if we don't need it, there is no reason to initialize this object?

If that were the case, then one should probably also (i) do the same change for the previous `diffusion_creep` and `dislocation_creep` objects or, (i) maybe more elegantly, make these objects `std::unique_ptr` objects that really only point to anything if they are used -- initialization would then happen at the place where you create the objects.

Regardless, this is a question separate from moving the one line of code above, so maybe we could decouple these into separate PRs.

Reply:
In order to not delay this PR (we currently have tests that run into floating point exceptions, although the tester does not pick them up) @naliboff: can you just revert this to the previous state? `frank_kamenetskii` is a valid member of this class, and as @bangerth points out there is no harm in calling these functions, even if the object is never used.

Reply:
Reverted :) I had guarded other calls elsewhere in visco_plastic, but now I don't recall why I did that. I any case, it was clearly unnecessary.

Reply:
Ah, I see now why you guarded the call. Now the tests fail because the rheology checks for the reference temperature being larger than 0. Can you go with the proper fix that Wolfgang proposed and make `frank_kamenetskii` a `std::unique_ptr` that only gets initialized if selected?

Reply:
👍 Working on this now. To be fair, I did not have that Assert in mind when guarding the calls :)

Reply:
Updated, let's see if this works. Thanks @gassmoeller for the explanation on why making the `frank_kamenetskii` a  `unique_ptr` is better practice.